server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -1902,6 +1911,12 @@ private void failSnapshotCompletionListeners(Snapshot snapshot, Exception e) {
         assert repositoryOperations.assertNotQueued(snapshot);
     }
 
+    private void failSnapshotCompletionListeners(Snapshot snapshot, Exception e, Consumer<Runnable> failingListenersConsumer) {
```

Comment:
Nit: maybe reimplement the other `failSnapshotCompletionListeners` as `failSnapshotCompletionListeners(snapshot, e, Runnable::run)`. And possibly just inline it, adding the `Runnable::run` parameter to its callers.